tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs

```diff
@@ -85,15 +76,51 @@ public async Task ServerCertificateCustomValidationCallback_ApprovesRequestWithI
 			};
 
 			var client = new HttpClient (handler);
+			await client.GetStringAsync ("https://self-signed.badssl.com/");
 
-			try {
-				await client.GetStringAsync ("https://self-signed.badssl.com/");
-			} catch (Exception e) {
-				exception = e;
-			}
+			Assert.IsTrue (callbackHasBeenCalled, "custom validation callback hasn't been called");
+		}
+
+		[Test]
+		public async Task NoServerCertificateCustomValidationCallback_ThrowsWhenThereIsCertificateHostnameMismatch ()
+		{
+			var handler = new AndroidMessageHandler ();
+			var client = new HttpClient (handler);
+
+			await AssertRejectsRemoteCertificate (() => client.GetStringAsync ("https://wrong.host.badssl.com/"));
+		}
+
+		[Test]
+		public async Task ServerCertificateCustomValidationCallback_IgnoresCertificateHostnameMismatch ()
+		{
+			bool callbackHasBeenCalled = false;
+			SslPolicyErrors reportedErrors = SslPolicyErrors.None;
+
+			var handler = new AndroidMessageHandler {
+				ServerCertificateCustomValidationCallback = (request, cert, chain, errors) => {
+					callbackHasBeenCalled = true;
+					reportedErrors = errors;
+					return true;
+				}
+			};
+
+			var client = new HttpClient (handler);
+			await client.GetStringAsync ("https://wrong.host.badssl.com/");
 
 			Assert.IsTrue (callbackHasBeenCalled, "custom validation callback hasn't been called");
-			Assert.IsNull (exception, $"an exception was thrown: {exception}");
+			Assert.AreEqual (SslPolicyErrors.RemoteCertificateNameMismatch, reportedErrors & SslPolicyErrors.RemoteCertificateNameMismatch);
+		}
+
+		private async Task AssertRejectsRemoteCertificate (Func<Task> makeRequest)
+		{
+			try {
+				await makeRequest();
+			} catch (Java.IO.IOException) {
```

Comment:
This is concerning, for the same reasons [as above](https://github.com/xamarin/xamarin-android/pull/7246/files#r950490896): [`HttpClient.GetStringAsync(string)`](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getstringasync?view=net-6.0#system-net-http-httpclient-getstringasync(system-string)) does not document `Java.IO.IOException` as an acceptable exception type, just: `InvalidOperationException`, `HttpRequestException`, and `TaskCanceledException`.

Consequently, `Java.IO.IOException` *should not be thrown*, ***ever***, from an `HttpClient.GetStringAsync()` invocation.

If a `Java.IO.IOException` can be thrown, *we must fix that*.